### PR TITLE
Add hitTolerance option to all annotations

### LINKED ISSUE #900: Configuring hitRadius for annotations
### Discussed in https://github.com/chartjs/chartjs-plugin-annotation/discussions/899

<div type='discussions-op-text'>

<sup>Originally posted by **aafanasev-shell** July  4, 2023</sup>
Hello!

In chart js therre is a possibility to configure `hitRadius` for data points which works like making "hitboxes" bigger. Is there anything similar for annotations in this plugin? 
I didn't find any option that can solve this issue - I understand that I probably can create custom `interaction mode` or override global onHover handler for Chart object, but it looks like there probably should be simple way to manage this through options like it's done in Chart js?

(line charts, line annotations)</div>

### PR DESCRIPTION
**Pull request by stockiNail**
Fix  #900

This PR is add new options, `hitTolerance`, which is a number and represents the amount of pixels to add for interacting on annotation elements.

### TODO

- [x] test case
- [x] types
- [x] documentation

### CODE REVIEW

**Comment by aafanasev-shell** [comment]
Hello! Can I somehow help with moving this to production? 🙂

**Comment by aafanasev-shell** [comment]
ping :)

**Comment by Nuffic** [comment]
I'm interested in this as well

File: src/helpers/helpers.core.js [hunk 1]
```diff
@@ -26,28 +33,27 @@ export function clampAll(obj, from, to) {
...
- * @param {number} borderWidth
+ * @param {{borderWidth: number, hitTolerance: number}} hitsize
  * @returns {boolean}
  */
-export function inBoxRange(point, {x, y, x2, y2}, axis, borderWidth) {
-  const hBorderWidth = borderWidth / 2;
-  const inRangeX = point.x >= x - hBorderWidth - EPSILON && point.x <= x2 + hBorderWidth + EPSILON;
-  const inRangeY = point.y >= y - hBorderWidth - EPSILON && point.y <= y2 + hBorderWidth + EPSILON;
+export function inBoxRange(point, {x, y, x2, y2}, axis, {borderWidth, hitTolerance}) {
+  const hitSize = borderWidth / 2 + hitTolerance / 2;
```
Comment (LeeLenaleee) [inline src/helpers/helpers.core.js:54]: Personally I find this a bit clearer/more readable and I se that you also do it like this later on

```suggestion
  const hitSize = (borderWidth + hitTolerance) / 2;
```

Comment (stockiNail) [reply src/helpers/helpers.core.js:54]: Thanks @LeeLenaleee . I'm going to apply your reviews (on other PRs) asap. Unfortunately these PRs are quite old and I don't remember well all code I wrote. :(

**Comment by stockiNail** [comment]
@LeeLenaleee ready for your review. I have also changed the tolerance for fixture/label/border.js test from 55 to 57. The test is correct but the small difference should be caused by new browser version.

**Review by LeeLenaleee — APPROVED** [review]

**Pull request merged**